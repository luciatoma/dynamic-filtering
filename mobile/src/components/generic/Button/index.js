import React from 'react';
import { StyleSheet, Text, TouchableHighlight } from 'react-native';
import PropTypes from 'prop-types';
import colors from '../../../styles/constants';

// Generic styled button
const ButtonStyled = props => {
    const { active, handleBtnClick, title } = props;

    const handleToggle = () => {
        handleBtnClick && handleBtnClick(!active, title);
    };

    return (
        <TouchableHighlight style={[active && styles.activeBtnWrapper]} onPress={handleToggle}>
            <Text style={[styles.button, active && styles.activeBtn]}>{title}</Text>
        </TouchableHighlight>
    );
};

const styles = StyleSheet.create({
    button: {
        borderColor: colors.blue,
        borderStyle: 'solid',
        borderWidth: 1,
        borderRadius: 4,
        backgroundColor: 'transparent',
        color: colors.blue,
        fontSize: 14,
        fontWeight: '500',
        paddingTop: 6,
        paddingRight: 6,
        paddingBottom: 6,
        paddingLeft: 6,
        textAlign: 'center',
    },
    activeBtnWrapper: {
        backgroundColor: colors.blue,
    },
    activeBtn: {
        color: colors.white,
    },
});

ButtonStyled.propTypes = {
    active: PropTypes.bool,
    handleBtnClick: PropTypes.func,
    title: PropTypes.string.isRequired,
};

export default ButtonStyled;
