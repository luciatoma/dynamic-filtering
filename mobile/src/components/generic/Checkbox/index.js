import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import CheckBox from 'react-native-check-box';

import colors from '../../../styles/constants';

// Generic styled checkbox input
const CheckboxStyled = props => {
    const { checked, handleCheck, title } = props;
    return (
        <CheckBox
            leftTextStyle={styles.checkboxText}
            leftText={title}
            checkBoxColor="white"
            isChecked={checked}
            onClick={() => handleCheck(!checked, title)}
        />
    );
};

const styles = StyleSheet.create({
    checkboxText: {
        color: colors.white,
    },
});
CheckboxStyled.propTypes = {
    checked: PropTypes.bool,
    handleCheck: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
};

export default CheckboxStyled;
