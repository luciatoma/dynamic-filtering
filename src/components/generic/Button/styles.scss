@import '../../../styles/constants.scss';

:local(.button) {
    background: transparent;
    border: 1px solid $blue;
    border-radius: 4px;
    color: $blue;
    cursor: pointer;
    font-size: 14px;
    font-weight: 500;
    min-width: 135px;
    padding: 6px;
    width: fit-content;

    &:focus {
        outline: none;
    }
}

:local(.active) {
    background: $blue;
    color: $white;
}

@media #{$tablet} {
    :local(.button) {
        width: 100%;
    }
}
