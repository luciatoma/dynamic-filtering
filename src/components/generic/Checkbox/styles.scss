@import '../../../styles//constants.scss';

:local(.check) {
    color: $white;
    cursor: pointer;
    display: block;
    font-size: 14px;
    margin-bottom: 12px;
    padding-right: 70px;
    position: relative;
    user-select: none;

    input {
        cursor: pointer;
        height: 0;
        opacity: 0;
        position: absolute;
        width: 0;
    }

    &:hover {
        input ~ :local(.checkmark) {
            background-color: $white;
        }
    }
}

:local(.checkmark) {
    background-color: transparent;
    border: 1px solid $white;
    border-radius: 4px;
    height: 16px;
    position: absolute;
    right: 0;
    width: 16px;

    &:after {
        border: solid $white;
        border-width: 0 2px 2px 0;
        content: '';
        display: none;
        height: 10px;
        left: 5px;
        position: absolute;
        top: 0px;
        transform: rotate(45deg);
        width: 4px;
    }
}

:local(.check) input:checked ~ :local(.checkmark) {
    background-color: $blue;

    &:after {
        display: block;
    }
}
