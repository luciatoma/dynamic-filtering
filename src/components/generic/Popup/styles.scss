@import '../../../styles/constants.scss';

:local(.wrapper) {
    background: $blue;
    border: 1px solid $blue;
    border-radius: 4px;
    box-shadow: 1px 0px 15px -5px rgba(0, 0, 0, 0.75);
    color: $white;
    margin-top: 5px;
    padding: 25px;
    position: absolute;
}

:local(.filterTitle) {
    font-weight: 600;
}

:local(.valuesWrapper) {
    border-top: 1px solid $white;
    margin: 10px 0 20px;
    padding-top: 10px;
}

.flatpickr-input {
    border: 1px solid $blue;
    border-radius: 4px;
    font-size: 12px;
    padding: 4px 6px;
    width: 100%;

    &:hover {
        border-color: $blue;
    }

    &:focus {
        outline: none;
        border-color: $blue;
        box-shadow: 0 0 0 2px $blue;
    }
}

.flatpickr-calendar {
    box-shadow: 1px 0px 15px -5px rgba(0, 0, 0, 0.75);

    .flatpickr-months .flatpickr-month,
    .flatpickr-innerContainer .flatpickr-weekdays,
    .flatpickr-innerContainer .flatpickr-weekday,
    .flatpickr-day.selected {
        background-color: $blue;
        border-color: $blue;
        color: $white;
    }
}
