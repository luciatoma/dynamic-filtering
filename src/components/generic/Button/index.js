import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './styles.scss';

const ButtonStyled = props => {
    const { active, handleBtnClick, title } = props;

    const handleToggle = () => {
        handleBtnClick && handleBtnClick(!active, title);
    };

    return (
        <button
            type="button"
            className={classNames({
                [styles.button]: true,
                [styles.active]: active,
            })}
            onClick={handleToggle}
        >
            {title}
        </button>
    );
};

ButtonStyled.propTypes = {
    active: PropTypes.bool,
    handleBtnClick: PropTypes.func,
    title: PropTypes.string.isRequired,
};

export default ButtonStyled;
