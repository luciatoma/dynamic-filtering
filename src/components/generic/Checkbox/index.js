import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scss';

// Generic styled checkbox input
const Checkbox = props => {
    const { value, onChange, title, checked } = props;
    return (
        <label className={styles.check}>
            {title}
            <input type="checkbox" value={value} defaultChecked={checked} onChange={onChange} />
            <span className={styles.checkmark} />
        </label>
    );
};

Checkbox.propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func,
    title: PropTypes.string.isRequired,
    checked: PropTypes.bool,
};

export default Checkbox;
