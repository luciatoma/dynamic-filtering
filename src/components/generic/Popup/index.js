import React from 'react';
import PropTypes from 'prop-types';
import Flatpickr from 'react-flatpickr';
import 'flatpickr/dist/themes/material_green.css';

import Checkbox from '../Checkbox';
import styles from './styles.scss';

const Popup = props => {
    const { activeFilters = {}, popupValues, title, userInput } = props;

    const handleCheckbox = (e, item) => {
        const {
            target: { value, checked },
        } = e;

        console.log('active', value);

        let values = activeFilters ? activeFilters[item] : [];
        if (!values) values = [];
        if (checked && !values.includes(value)) values.push(value);
        if (!checked && values.includes(value)) values = values.filter(option => option !== value);

        const newChecked = { ...activeFilters, [item]: values };

        popupValues && popupValues(newChecked, title);
    };

    const handleDatePickr = (date, item) => {
        const newChecked = { ...activeFilters, [item]: date };

        popupValues && popupValues(newChecked, title);
    };

    return (
        <div className={styles.wrapper}>
            {userInput.map(item => {
                if (item.filter !== 'StartDate') {
                    return (
                        <div key={item.filter}>
                            <span className={styles.filterTitle}>Select {item.filter}</span>
                            <div className={styles.valuesWrapper}>
                                {item.values.map(value => (
                                    <Checkbox
                                        key={value}
                                        title={value}
                                        value={value}
                                        checked={activeFilters[item.filter] && activeFilters[item.filter].includes(value)}
                                        onChange={e => handleCheckbox(e, item.filter)}
                                    />
                                ))}
                            </div>
                        </div>
                    );
                }

                return (
                    <div key={item.filter}>
                        <span className={styles.filterTitle}>Select {item.filter}</span>
                        <Flatpickr
                            value={activeFilters[item.filter] || new Date()}
                            onChange={date => handleDatePickr(date, item.filter)}
                        />
                    </div>
                );
            })}
        </div>
    );
};

Popup.propTypes = {
    activeFilters: PropTypes.object,
    popupValues: PropTypes.func,
    title: PropTypes.string,
    userInput: PropTypes.array,
};

export default Popup;
