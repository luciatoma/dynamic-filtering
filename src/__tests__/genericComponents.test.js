import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Button from '../components/generic/Button';
import Checkbox from '../components/generic/Checkbox';
import Popup from '../components/generic/Popup';
import WorkOrder from '../components/generic/WorkOrderCard';

/* global document */

// keeping the code dry
const tests = [
    { name: 'Button', component: <Button title="test" /> },
    { name: 'Checkbox', component: <Checkbox title="test" value="test" /> },
    {
        name: 'Popup',
        component: (
            <Popup popupValues={jest.fn} userInput={[{ filter: 'test', values: ['test'] }]} activeFilters={{ test: [] }} />
        ),
    },
    {
        name: 'WorkOrder',
        component: (
            <WorkOrder
                description="test"
                color="blue"
                endDate="test"
                name="test"
                startDate="test"
                status="Open"
                type="Maintenance"
            />
        ),
    },
];

tests.forEach(test =>
    it(`${test.name} renders without crashing and matches snapshot`, () => {
        const div = document.createElement('div');
        ReactDOM.render(test.component, div);
        ReactDOM.unmountComponentAtNode(div);
        const tree = renderer.create(test.component).toJSON();
        expect(tree).toMatchSnapshot();
    })
);
