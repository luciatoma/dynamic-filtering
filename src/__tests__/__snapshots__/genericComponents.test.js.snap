// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing and matches snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Checkbox renders without crashing and matches snapshot 1`] = `
<label>
  test
  <input
    type="checkbox"
    value="test"
  />
  <span />
</label>
`;

exports[`Popup renders without crashing and matches snapshot 1`] = `
<div>
  <div>
    <span>
      Select 
      test
    </span>
    <div>
      <label>
        test
        <input
          defaultChecked={false}
          onChange={[Function]}
          type="checkbox"
          value="test"
        />
        <span />
      </label>
    </div>
  </div>
</div>
`;

exports[`WorkOrder renders without crashing and matches snapshot 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <span>
      Invalid date
       -
       
      Invalid date
    </span>
    <span>
      test
    </span>
    <span
      className="undefined"
    >
      <span>
        Maintenance
      </span>
      <span>
        Status: 
        Open
      </span>
    </span>
  </div>
  <div>
    <div>
      <span>
        Customer
      </span>
      <span>
        Magna
      </span>
      <span>
        Meijendel 26
      </span>
      <span>
        1941CM Beverwijk
      </span>
    </div>
    <div>
      <span>
        Installation
      </span>
      <span>
        Suntech 245Wp-2151
      </span>
      <span>
        Contract
      </span>
      <span>
        Full Service 24/7
      </span>
    </div>
    <div>
      <span>
        Notification
      </span>
      <span>
        solar panel does not work
      </span>
      <span>
        Office remarks
      </span>
      <span>
        test
      </span>
    </div>
  </div>
</div>
`;
